x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/signals_alert_type.ts

```diff
@@ -79,20 +99,120 @@ export const signalsAlertType = ({ logger }: { logger: Logger }): AlertType => {
+        try {
+          const nextScrollResult = await services.callCluster('scroll', {
+            scroll,
+            scrollId,
+          });
+          return nextScrollResult;
+        } catch (exc) {
+          logger.error(`[-] nextScroll threw an error ${JSON.stringify(exc)}`);
+          throw exc;
+        }
+      };
+
+      async function bulkIndex(someResult: SearchResponse<object>): Promise<boolean> {
+        logger.info('[+] starting bulk insertion');
+        const firstBulkIndexSuccess = await singleBulkIndex(someResult);
+        if (!firstBulkIndexSuccess) {
+          logger.error('[-] First bulk index threw an error');
+          return false;
+        }
+        while (true) {
+          try {
+            // reusing scroll id from initial call
+            const scrollResult = await singleScroll(someResult._scroll_id);
+            if (scrollResult.hits.hits.length === 0) {
+              logger.info('[+] Finished indexing signals');
+              return true;
+            }
+            const bulkSuccess = await singleBulkIndex(scrollResult);
+            if (!bulkSuccess) {
+              logger.error('[-] bulk index failed');
+            }
+          } catch (exc) {
+            logger.error('[-] scroll and bulk threw an error');
+            return false;
+          }
+        }
+      }
+
       try {
         logger.info('Starting SIEM signal job');
 
         // TODO: Comment this in eventually and use this for manual insertion of the
         // signals instead of the ReIndex() api
-        // const result = await services.callCluster('search', query);
-        const result = await services.callCluster('reindex', reIndex);
 
-        // TODO: Error handling here and writing of any errors that come back from ES by
-        logger.info(`Result of reindex: ${JSON.stringify(result, null, 2)}`);
+        if (process.env.USE_REINDEX_API === 'USE_REINDEX_API') {
+          // eslint-disable-next-line
+          const result = await services.callCluster('reindex', reIndex);
+
+          // TODO: Error handling here and writing of any errors that come back from ES by
+          logger.info(`Result of reindex: ${JSON.stringify(result, null, 2)}`);
+        } else {
+          logger.info(`[+] Initial search call`);
+          const noReIndexResult = await services.callCluster('search', noReIndex);
+          logger.info(`Total docs to reindex: ${noReIndexResult.hits.total.value}`);
+          const bulkIndexResult = await bulkIndex(noReIndexResult);
+          if (bulkIndexResult) {
+            logger.info('Finished SIEM signal job');
+          } else {
+            logger.info('Error processing SIEM signal job');
+          }
+        }
       } catch (err) {
         // TODO: Error handling and writing of errors into a signal that has error
         // handling/conditions
-        logger.error(`You encountered an error of: ${err.message}`);
+        logger.info(`You encountered an error of: ${err.message}`);
```

Comment:
Did you want to change this from an `error` to an `info`? It looks like it is outputting when there is an error.

Reply:
Ah yeah that was a mistake. Thank you.